src/routes/breaddit/breaddit.controller.js

```diff
@@ -0,0 +1,104 @@
+const fs = require("fs");
+const { promisify } = require("util");
+const { validationResult } = require("express-validator");
+
+const breadditsData = require("../../../db/dummy.data.json");
+const data = breadditsData.breaddits;
+
+const writeFile = promisify(fs.writeFile);
+
+const listBreaddits = (req, res) => {
+  return res.json({
+    data
+  });
+};
+
+const createBreaddit = async (req, res) => {
+  // console.log('body', req.body)
+  const id = data.length + 100;
+  const errors = validationResult(req);
+  if (!errors.isEmpty()) {
+    return res.status(422).json({ errors: errors.array() });
+  }
+  const newData = {
+    title: data.title,
+    breaddits: [...data, { id, ...req.body }]
+  };
+
+  await writeFile("db/breaddits-copy.data.json", JSON.stringify(newData));
+  return res.status(201).json({
+    id,
+    ...req.body
+  });
+}
+
+// GET show id
+const getBreaddit = async (req, res) => {
+  const { id } = req.params;
+  const getBreadditById = data.find(breaddit => breaddit.id == id);
+  if (getBreadditById) {
+    return res.status(200).json({
+      message: "One breaddit found",
+      getBreadditById
+    });
+  } else {
+    res.status(400).json({
+      error: "No breaddit found with that id"
+    })
+  }
+};
+
+// DELETE a specific breaddit
+const deleteBreaddit = async (req, res) => {
+  let { id } = req.params;
+  const getBreadditById = data.find(breaddit => breaddit.id == id);
+  if (getBreadditById) {
+    const newData = data.filter(breaddits => {
+      return breaddits !== getBreadditById;
+    })
+
+    const updatedData = {
+      breaddits: newData
+    }
+    await writeFile("db/breaddits-copy.data.json", JSON.stringify(updatedData));
+    res.status(200).json({
+      message: "Breaddit successfully deleted!",
+      data: newData
+    });
+  } else {
+    res.status(400).json({
+      error: "Could not delete this breaddit."
+    })
+  }
+}
+
+// PATCH - update a breaddit
+const updateBreaddit = async (req, res) => {
+  const { id } = req.params;
+  const getBreadditById = data.find(breaddit => breaddit.id == id);
+  if (getBreadditById) {
+    (getBreadditById.title = req.body.title), (getBreadditById.description = req.body.description);
+    console.log({ getBreadditById });
+
+    const newData = {
+      breaddits: [...data]
+    }
+    await writeFile("db/breaddits-copy.data.json", JSON.stringify(newData));
```

Comment:
Curious: What was the reasoning behind writing the updated data to a new file? The next time a request comes in, the previous changes won't have been persisted. Though, perhaps a moot point since I hear y'all will be plugging into an actual database soon :-)

Reply:
I was actually working with this dummy data file as it has fewer items in it so easier to see changes in the list.  I have a task to clean this up but it is a low priority task as I was sure we would be changing up the database aspect in the coming weeks.